pyocd/probe/stlink/usb.py

```diff
@@ -126,8 +126,11 @@ def __init__(self, dev):
         # reading the strings, everything is ok. This workaround doesn't cause any issues with
         # Linux or macOS.
         try:
-            self._serial_number = self._dev.serial_number
-            self._vendor_name =  self._dev.manufacturer
+            if self._dev.idProduct == 0x3748:  # Workaround for unprintable characters in the ST-Link V2 probes
+                self._serial_number = bytes(self._dev.serial_number, 'utf-16')[2::2].hex().upper()
```

Comment:
Unfortunately this line won't work for Python 2.7.

Reply:
Sorry, completely missed it. Is it better now?

Reply:
Looks good!